rust/basic_bitcoin/src/basic_bitcoin/src/bitcoin_wallet.rs

```diff
@@ -204,20 +249,47 @@ async fn sign_transaction(
         sig_with_hashtype.push(SIG_HASH_TYPE.as_u32() as u8);
         input.script_sig = Builder::new()
             .push_slice(sig_with_hashtype.as_slice())
-            .push_slice(public_key.as_slice())
+            .push_slice(own_public_key)
             .into_script();
         input.witness.clear();
     }
 
     transaction
 }
 
-fn sha256(data: Vec<u8>) -> Vec<u8> {
+fn sha256(data: &[u8]) -> Vec<u8> {
     let mut hasher = sha2::Sha256::new();
     hasher.update(data);
     hasher.finalize().to_vec()
 }
 
+// Converts a public key to a P2PKH address.
+fn public_key_to_p2pkh_address(network: Network, public_key: &[u8]) -> String {
```

Comment:
Just using [`Address::p2pkh` from the Bitcoin crate](https://docs.rs/bitcoin/0.28.1/bitcoin/util/address/struct.Address.html#method.p2pkh) may be simpler.

Reply:
I'd like to stick with the primitives. One benefit of that is that this code will match the logic written in Motoko.